include/range/v3/view_interface.hpp

```diff
@@ -297,6 +297,19 @@ namespace ranges
             {
                 return Slice{}(detail::move(derived()), offs.from, offs.to);
             }
+            /// Returns a reference to the element at specified location pos, with bounds checking.
+            template<typename D = Derived,
+                CONCEPT_REQUIRES_(Same<D, Derived>() && RandomAccessRange<D>())>
+            auto at( range_difference_type_t<D> n) ->
+                decltype(std::declval<D &>().begin()[n])
+            {
+                using size_type = decltype( derived().size() );
+                if ( (n < 0) or
+                     (size_type(n) >= derived().size()) ){
+                  throw std::out_of_range("view_interface::at");
+                }
+                return derived()[n];
```

Comment:
`RandomAccessRange` doesn't require a `[]` operator or require that `derived()[n]` has the same type as `derived().begin()[n]`; this should return `derived().begin()[n]`. (A user may have overriden or disabled `[]` in `Derived`.)